### PR TITLE
Additional: cifES

### PR DESCRIPTION
**Pull request by egprida**
Bug when codes beginning with lower case letter.

#### Description

Thank you!

### CODE REVIEW

**Comment by RobJohnston** [comment]
I went to the Wikipedia link provided in the source code, and then to the referenced documents at the "Agencia Estatal Boletín Oficial del Estado".  Although all the examples show an upper-case letter, I saw nothing that said it was **required** to be upper-case.  

Full discretion, I'm functional in Spanish, and use it almost every day in some way, but I'm far from fluent.  I stand to be corrected.

**Comment by jarey** [comment]
Hello @RobJohnston ,

As you said is not required that the NIF uses upper-case letters only.
The problem with the actual code (and wich this pull request solves) is that the current regexp only works with upper case letters.
If you see the detail of the PR, the change only applies upper-case transformation to the NIF letter in order to match the existent regexp and solve the validation problem that exists in case that is specified a NIF with lower-case letter.

Regards.

File: test/methods.js [hunk 1]
```diff
@@ -1555,7 +1555,7 @@ QUnit.test( "cifES", function( assert ) {
 	assert.ok( method( "S5141387J" ), "CIF valid" );
 	assert.ok( method( "S9803881C" ), "CIF valid" );
 	assert.ok( !method( "S98038813" ), "CIF invalid: digit control must be a letter (C)" );
-	assert.ok( method( "s98038813" ), "CIF valid: lower case" );
+	assert.ok( !method( "s98038813" ), "CIF invalid: lower case but digit control must be a letter (C)" );
```
Comment (staabm) [inline test/methods.js:1558]: why is this considered invalid now?

in your comment you said lower case chars are fine.
what das "but digit control must be a letter" mean?

File: src/additional/cifES.js [hunk 2]
```diff
@@ -55,6 +55,8 @@ $.validator.addMethod( "cifES", function( value, element ) {
 		return true;
 	}
 
+	value = value.toUpperCase();
```
Comment (staabm) [inline src/additional/cifES.js:58]:
```suggestion
    // turn the validation case-insensitive
	value = value.toUpperCase();
```

**Pull request closed without merging**